### ISSUE TITLE
Better error handling

### ISSUE THREAD

**Issue by josefdolezal** [sunshinejr/Moya-ModelMapper#38]
Hi @sunshinejr! While using this awesome library (now with Swift PM, haha), we sometimes have a really hard times debugging the mapping errors. Usually when the backend API spec is not clear about types or fields requirement, the mapping fails. When this happens, the Mapper throws message like this:

```
jsonMapping(Status Code: 200, Data Length: 11762)
``` 

Now it's pretty hard to say what and why happened - you have to use debugger and/or debug prints. Yay!

I propose small update of Mapper extension to rethrow Mapper errors as associated values of `MoyaError.underlaying`.

The code (referencing `Response+ModelMapper.swift`) would look like:

```swift
public func mapObject<T: Mappable>() throws -> T {
    guard let jsonDictionary = try mapJSON() as? NSDictionary else {
        throw MoyaError.jsonMapping(self)
    }

    do {
        return try T(map: Mapper(JSON: jsonDictionary))
    } catch {
        throw MoyaError.underlying(error)
    }
}
```

Which is just a little bit more complex but gives nice and clear errors:

```
underlying(Mapper.MapperError.missingFieldError("tags"))
```

The change is not breaking in sense of source compatibility, but requires to observe for both `.jsonMapping` and `.underlying` errors. Would you be interested in PR?

**Comment by sunshinejr** [comment]
Hey @josefdolezal, thank you so much for this issue! I love this idea, I must say I must've missed the release where this mapping errors came out, but let's get it in the next version!

Would love it if you could make a PR with changes proposed here. I don't have that much free time right now 🐼

**Comment by josefdolezal** [comment]
Thanks for your kind answer! I've opened the PR #39 with proposed changes.

Please take all the time you need, no need to hurry!

**Comment by sunshinejr** [comment]
Fixed by #39.
